### PR TITLE
Upgrade to go1.15.5

### PR DESCRIPTION
**Pull request by SilvaMatteus**
Signed-off-by: silvamatteus <silvamatteus@lsd.ufcg.edu.br>

**Pull Request checklist**

- [X] Commit conforms to CONTRIBUTING.md?
- [X] Proper tests/regressions included?
- [ ] Documentation updated?

**Description of change**

Upgrade go version to 1.15.5.

**Which issue this PR fixes**

### CODE REVIEW

**Comment by azdagron** [comment]
Hi @SilvaMatteus ! Thanks for opening this. The PR description says that this also moves the code over to use the T.TempDir method but I don't see that?

Also, it looks like there have been some subtle changes in the Go1.15 stdlib that is causing some of our unit tests to fail...

**Comment by SilvaMatteus** [comment]
Hi @azdagron !

It's still a WIP. I used the option "create a draft pull request". Let me know if this not the best approach for works in progress.

I started, but the time was short. Actually, I updated the tests that broke with the upgrade to 1.15.4 already. I'll push the changes now so you can verify if my solution is fine.

**Comment by azdagron** [comment]
Whoops! Didn't see it was a draft. Carry on.... nothing to see here :)

**Comment by SilvaMatteus** [comment]
@azdagron, it's ready to review now.

**Comment by azdagron** [comment]
If we're going to fix up go.mod, we should probably also move to go1.15.5, which came out yesterday 😆

File: .go-version [hunk 1]
```diff
@@ -1 +1 @@
-1.14.6
+1.15.4
```
[the remark this answers is not in the document]
Comment (SilvaMatteus) [reply .go-version:1]: Hey @MarcosDY, thanks for catching this!

**Comment by SilvaMatteus** [comment]
@azdagron great! Moving to go1.15.5 then :)

**Comment by azdagron** [comment]
I think we're running up against macOS limitations on unix domain socket path lengths in the unit tests...

From the UNIX(4) man page:
> UNIX-domain addresses are variable-length filesystem pathnames of at most 104 characters

**Comment by azdagron** [comment]
The difference between our TempDir helper and the golang stdlib one is that they include the test function name, which is some cases is causing it to exceed the UDS length limitation. I don't have a good solution here. We may need to roll back the changes to use t.TempDir() unless somebody can think of something. The alternative is shortening the test function names, but that feels a little gross to me. It would be unfortunate to place arbitrary restrictions on test authors.

**Comment by azdagron** [comment]
Talked to @SilvaMatteus offline. We're just going to take the upgrade to go1.15 right now and figure out what, if anything to do about the TempDir() path length. Might have to just stick to our helper instead of using Go's.

**Review by azdagron — APPROVED** [review]

**Pull request merged**